app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt

```diff
@@ -152,10 +156,11 @@ class VisitsListFragment(private val visitsKey: String) : BaseFragment(),
     }
 
     private fun exportToExcel(visits: List<VisitDataDTO>) {
-        val fileName = "${buildFileName()}.xls"
-        val mimeType = "application/vnd.ms-excel"
+        val fileName = "${buildFileName()}.xlsx"
+        val mimeType = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+        val file = File(requireContext().getExternalFilesDir(null), fileName)  // Use external storage
 
-        FileUtil.exportToFile(requireContext(), fileName, mimeType) { outputStream ->
```

Comment:
please don't change it. If you change you have to create a new method that you created below for openExcelFile. This is almost the same method that you have in FileUtil for opening files. 

You should use only exportToFile everywhere, and openFile method from FileUtil - that was the reason to create FileUtil class - to not duplicate the same methods multiple times.